controllers/machine_controller.go

```diff
@@ -228,6 +229,8 @@ func patchMachine(ctx context.Context, patchHelper *patch.Helper, machine *clust
 			clusterv1.BootstrapReadyCondition,
 			clusterv1.InfrastructureReadyCondition,
 			clusterv1.DrainingSucceededCondition,
+			clusterv1.MachineOwnerRemediatedCondition,
+			clusterv1.MachineHealthCheckSuccededCondition,
```

Comment:
```suggestion
```
MHC should not go there because the idea is to have a step counter only during machine provisioning, but to get it removed when machines is running

Reply:
They are not added to the step counter, but to the `patchHelper`

Reply:
Sorry, got tricked by git skipping lines